apisix/plugins/proxy-rewrite.lua

```diff
@@ -178,6 +183,28 @@ function _M.rewrite(conf, ctx)
         ctx.var.upstream_uri = upstream_uri
     end
 
+    -- enum    = {"GET", "POST","PUT","DELETE"}
+    -- change Http method
+    if conf.method ~=nil then
+
+      if conf.method == "GET" then
+        ngx.req.set_method(ngx.HTTP_GET)
+
+      elseif conf.method == "POST" then
+        ngx.req.set_method(ngx.HTTP_POST)
+
+      elseif conf.method == "PUT" then
+        ngx.req.set_method(ngx.HTTP_PUT)
+
+      elseif conf.method == "DELETE" then
+        ngx.req.set_method(ngx.HTTP_DELETE)
+      else
```

Comment:
Seems the if else judgements are not complete, the `PATCH` method should also be supported.